docs/modules/record-encryption/con-lost-kek.adoc

```diff
@@ -0,0 +1,94 @@
+// file included in the following:
+//
+// assemblies/assembly-operations-record-encryption-filter.adoc
+
+[id='con-lost-kek-{context}']
+= Dealing with a lost KEK
+
+This section discusses what to do if KEK that is still in use to encrypt topics is KEK is lost.
+
+WARNING: It is stressed that for successful use of the Record Encryption filter that the encryption keys are *never* deleted from the KMS.
+These procedures are provided for emergency use only.
```

Comment:
This is not really correct. It's fine to delete a KEK if there are no remaining records whose DEKs were encrypted with that KEK. The real problem is that, given Kafka retention semantics, that's hard to figure out, and the answer can indeed be "never", but it isn't always.

```suggestion
WARNING: Deleting KEKs from the KMS is not recommended. 
Figuring out which KEKs are no longer used by any records in a Kafka cluster is non-trivial.
Any remaining records encrypted using a deleted KEK will become *unrecoverable*. 
In addition, consumers will be unable to read beyond those unrecoverable records without specific intervention, meaning business processing will stop.
We recommend never deleting KEKs. As such, this procedure is provided for emergency use only.
```